### PR TITLE
feat(j-s): Context menu

### PR DESCRIPTION
**Pull request by oddsson**
# Send case files functionality

[Asana](https://app.asana.com/0/1199153462262248/1206195048528166/f)

## What

Change how prosecutors and defenders send additional case files to appeal cases. Now, case files are not uploaded until you press the "Send case files" button. The case files can now be deleted in the case files overview. 

## Why

We believe it's clearer for users that files are uploaded in this pattern.

## Screenshots / Gifs

https://github.com/island-is/island.is/assets/3789875/28a6ac3e-809a-4e61-ada6-ae5ef0a04e2b

### Upload fails

https://github.com/island-is/island.is/assets/3789875/4773eb71-55d0-404c-a648-73e3438b6781

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Shared/AppealToCourtOfAppeals/AppealCaseFiles.tsx [hunk 1]
```diff
@@ -156,16 +195,20 @@ const AppealFiles = () => {
...
+              : strings.nextButtonText,
+          )}
           nextButtonIcon={undefined}
+          nextIsLoading={uploadState.isUploading}
+          nextIsDisabled={newFiles.length === 0}
+          nextButtonColorScheme={
+            uploadState.error && !uploadState.isUploading && newFiles.length > 0
+              ? 'destructive'
+              : 'default'
+          }
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Shared/AppealToCourtOfAppeals/AppealCaseFiles.tsx:211]: I tried this locally and only a single file is uploaded even if I select multiple files. Not sure what is happening.

File: apps/judicial-system/web/src/utils/hooks/useS3Upload/useS3Upload.ts [hunk 2]
```diff
@@ -281,11 +287,14 @@ const useS3Upload = (caseId: string) => {
             // We need to set the id so we are able to delete the file later
             newFileId,
           )
+
+          return true
```
Comment (gudjong) [inline apps/judicial-system/web/src/utils/hooks/useS3Upload/useS3Upload.ts:291]: Returning from the loop means that we only try to upload the first file. The rest are ignored.

**Review by gudjong — CHANGES_REQUESTED** [review]
Some comments that need to be addressed.

File: apps/judicial-system/web/src/components/AppealCaseFilesOverview/AppealCaseFilesOverview.tsx [hunk 3]
```diff
@@ -106,35 +129,77 @@ const AppealCaseFilesOverview: React.FC<
...
+                      </Text>
+                      <Text variant="small">
+                        {formatMessage(strings.submittedBy, {
+                          filesCategory: prosecutorSubmitted,
+                        })}
+                      </Text>
+                    </Box>
+                    {((prosecutorSubmitted && isProsecutionUser(user)) ||
+                      (!prosecutorSubmitted && isDefenceUser(user))) && (
+                      <Box marginLeft={3}>
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/AppealCaseFilesOverview/AppealCaseFilesOverview.tsx]: Good catch! Thanks..

**Review by gudjong — APPROVED** [review]

**Pull request merged**